src/lucky_record/model.cr

```diff
@@ -103,4 +107,89 @@ class LuckyRecord::Model
   macro association(table_name, foreign_key = nil)
     {% ASSOCIATIONS << {name: table_name.id, foreign_key: foreign_key} %}
   end
+
+  macro setup_correct_field_mappings_method
+    def ensure_correct_field_mappings!
+      ensure_existing_table!
+
+      fields = [
+        {% for field in FIELDS %}
+          { name: :{{field[:name]}}, nilable: {{ field[:nilable] }}, type: {{ field[:type] }} },
+        {% end %}
+      ]
+      ensure_fields_match_columns! fields
+    end
+  end
+
+  def ensure_existing_table!
+    table_names = LuckyRecord::Repo.tables_with_schema(excluding: "migrations")
+    return if table_names.includes?(@@table_name.to_s)
+
+    best_match = Levenshtein::Finder.find @@table_name.to_s, table_names, tolerance: 4
+    message = "The table #{@@table_name} was not found"
+
+    if best_match
+      message += ", did you mean #{best_match}?"
+    end
+
+    raise message
+  end
+
+  def ensure_fields_match_columns!(fields)
+    columns_map = Hash(String, Bool).new
+    LuckyRecord::Repo.table_columns(@@table_name).each do |column|
+      columns_map[column.name] = column.nilable
+    end
+
+    missing_columns = [] of String
+    optional_field_errors = [] of String
+    required_field_errors = [] of String
+
+    fields.each do |field|
+      unless columns_map.has_key? field[:name].to_s
+        missing_columns << missing_field_error(field)
+        next
+      end
+
+      if !field[:nilable] && columns_map[field[:name].to_s]
+        required_field_errors << required_field_error(field)
+      elsif field[:nilable] && !columns_map[field[:name].to_s]
+        optional_field_errors << optional_field_error(field)
+      end
+    end
+
+    if missing_columns.empty? && optional_field_errors.empty? &&required_field_errors.empty?
+      return
+    end
+
+    message = missing_columns + optional_field_errors + required_field_errors
+
+    raise message.join("\n\n")
```

Comment:
I think this could also be extracted to a class in `LuckyRecord::SchemaEnforcer` called `EnsureMatchingColumns`

Then you could extract some private methods for some of these to keep the class super easy to read. LMK if you want some help with this.